src/lab.c

```diff
@@ -2040,6 +2046,31 @@ void CPUThink(GOBJ *event, GOBJ *hmn, GOBJ *cpu)
             int shieldCtr = LabOptions_CPU[OPTCPU_CTRSHIELD].option_val;
             action_id = CPUCounterActionsShield[shieldCtr];
         }
+
+        // Additional check for the grab release counter action
+        else if (cpu_data->TM.state_prev[0] == ASID_CAPTURECUT || cpu_data->TM.state_prev[0] == ASID_CAPTUREJUMP)
+        {
+
+            if (eventData->cpu_countertimer < LabOptions_CPU[OPTCPU_CTRFRAMES].option_val)
+            {
+                break;
+            }
+            
+            // This is needed to avoid the cpu going into COUNTER_STATE indefinitly when the counter action is set to NONE
+            cpu_data->TM.state_prev[0] = CPUSTATE_COUNTER;
```

Comment:
It's unfortunate that term state is overloaded here. There are two different 'state' types, the action state (ASID_*), and the CPU state (CPUSTATE_*). The state_prev in TM contains the action state, and not the cpu state.

What's more, state_prev is ostensibly read only, please do not set this, it's used by a bunch of other stuff and setting the value can mess with OSDs and what not.

I looked into the problem here, and the reason we enter this a bunch when the CPU is set to counter act None is because the condition on line 1234 is triggered every frame, because we never start another action.

So the solution here I think is to remove this line, then add another check to line 1234 that ensures that it is only true when the CPU is in the first frame of the action state after exiting the breakout animation - `cpu_data->TM.state_frame == 1`